src/Storages/StorageDistributed.cpp

```diff
@@ -700,6 +714,8 @@ ClusterPtr StorageDistributed::getOptimizedCluster(const Context & context, cons
         std::stringstream exception_message;
         if (!has_sharding_key)
             exception_message << "No sharding key";
+        else if (sharding_key_is_deterministic)
```

Comment:
Sorry, there is a typo, here `!sharding_key_is_deterministic` should be.
@alexey-milovidov can you commit this simple fix?

Reply:
@alexey-milovidov friendly ping